plugin/auth_ldap/src/auth_ldap_connection.cc

```diff
@@ -0,0 +1,288 @@
+
+  log_stream << "search_dn(" << base_dn << ", " << filter << ")";
+  log_srv_dbg(log_stream.str());
+  log_stream.str("");
+
+  LDAPMessage *res;
+  char *attrs[] = {strdup("dn"), nullptr};
+  struct timeval search_timeout;
+  memset(&search_timeout, 0, sizeof(struct timeval));
+  search_timeout.tv_sec = 5;
+  int searchlimit = 1;
+  int err = ldap_search_ext_s(
+      this->ldap, base_dn.c_str() /* base */, LDAP_SCOPE_SUBTREE /*scope*/,
+      filter.c_str() /*filter*/, attrs /*attrs*/, 0 /*attrsonly*/,
+      nullptr /*serverctrls*/, nullptr /*clientctrls*/,
+      &search_timeout /*timeout*/, searchlimit /*searchlimit*/,
+      &res /*ldapmessage*/);
+  if (err == LDAP_SUCCESS) {
+    // Verify an entry was found
+    if (ldap_count_entries(this->ldap, res) == 0) {
+      log_stream << "ldap_search_ext_s(" << base_dn << ", " << filter
+                 << ") returned no matching entries";
+      log_srv_warn(log_stream.str());
+      log_stream.str("");
+      // Only free up res if there are no items
+      ldap_msgfree(res);
+      res = nullptr;
+    } else {
+      LDAPMessage *entry = ldap_first_entry(this->ldap, res);
+      char *dn = ldap_get_dn(this->ldap, entry);
+      log_stream << "ldap_search_ext_s(" << base_dn << ", " << filter
+                 << "): " << dn;
+      log_srv_dbg(log_stream.str());
+      log_stream.str("");
+      str = dn;
+      ldap_memfree(dn);
+      ldap_memfree(entry);
+    }
+  } else {
+    log_stream << "ERROR: ldap_search_ext_s(" << base_dn << ", " << filter
+               << ") " << ldap_err2string(err);
+    log_srv_error(log_stream.str());
+    log_stream.str("");
+  }
+
+  log_stream << "search_dn(" << base_dn << ", " << filter << ") = " << str;
+  log_srv_dbg(log_stream.str());
+  log_stream.str("");
+
+  return str;
+}
+
+std::list<std::string> AuthLDAPConnection::search_groups(
+    std::string user_name, std::string user_dn, std::string group_search_attr,
+    std::string group_search_filter, std::string base_dn) {
+  std::list<std::string> list;
+  std::stringstream log_stream;
+  std::string filter = std::regex_replace(group_search_filter,
+                                          std::regex("\\{UA\\}"), user_name);
+  filter = std::regex_replace(filter, std::regex("\\{UD\\}"), user_dn);
+  log_stream << "search_groups() - filter: " << filter;
+  log_srv_dbg(log_stream.str());
+  log_stream.str("");
+
+  LDAPMessage *res;
+  char *attrs[] = {strdup(group_search_attr.c_str()), nullptr};
+  struct timeval search_timeout;
+  memset(&search_timeout, 0, sizeof(struct timeval));
+  search_timeout.tv_sec = 5;
+  int err = ldap_search_ext_s(
+      this->ldap, base_dn.c_str() /* base */, LDAP_SCOPE_SUBTREE /*scope*/,
+      filter.c_str() /*filter*/, attrs /*attrs*/, 0 /*attrsonly*/,
+      nullptr /*serverctrls*/, nullptr /*clientctrls*/,
+      &search_timeout /*timeout*/, 0 /*searchlimit*/, &res /*ldapmessage*/);
+  if (err == LDAP_SUCCESS) {
+    // Verify an entry was found
```

Comment:
Please use `/**/` for comments

Reply:
?? I can see multiple // within MySQL. Is this something upstream is trying to change?